python/ray/_private/structured_logging.py

```diff
@@ -0,0 +1,42 @@
+import logging
+import ray
+import json
+
+from ray._private.ray_constants import (
+    LOG_RAY_CORE_CONTEXT,
+    LOGGER_RAY_ATTRS,
+    LOGRECORD_STANDARD_ATTRS,
+)
+
+
+class CoreContextFilter(logging.Filter):
+    def filter(self, record):
+        runtime_context = ray.get_runtime_context()
+        ray_core_context = {
+            "job_id": runtime_context.get_job_id(),
+            "worker_id": runtime_context.get_worker_id(),
+            "node_id": runtime_context.get_node_id(),
+        }
+        if runtime_context.worker.mode == ray.WORKER_MODE:
+            ray_core_context["actor_id"] = runtime_context.get_actor_id()
+            ray_core_context["task_id"] = runtime_context.get_task_id()
```

Comment:
We should only set them if they are not None

Reply:
Update https://github.com/ray-project/ray/pull/45098/commits/b2b2196a31eb8a907c4ca73038acfe7ee1a33de4.